rllib/core/rl_trainer/trainer_runner.py

```diff
@@ -221,24 +279,30 @@ def remove_module(self, module_id: ModuleID) -> None:
         else:
             self._trainer.remove_module(module_id)
 
-    def get_weight(self) -> Dict:
-        """Get the weights of the MARLModule.
+    def set_weights(self, weights) -> None:
+        # TODO (Kourosh) Set / get weight has to be thoroughly
+        # tested across actors and multi-gpus
+        if self._distributed:
+            ray.get([worker.set_weights.remote(weights) for worker in self._workers])
+        else:
+            self._trainer.set_weights(weights)
 
-        Returns:
-            The weights of the neural networks that can be exchanged with the policy.
-        """
-        # TODO (Avnish): implement this.
-        pass
+    def get_weights(self, module_ids: Optional[Set[str]] = None) -> Mapping[str, Any]:
+        if self._distributed:
+            worker = next(iter(self._workers))
+            return ray.get(worker.get_weights.remote(module_ids))
+        else:
+            return self._trainer.get_weights(module_ids)
 
     def get_state(self) -> List[Mapping[ModuleID, Mapping[str, Any]]]:
         """Get the states of the RLTrainers"""
         if self._distributed:
             refs = []
             for worker in self._workers:
                 refs.append(worker.get_state.remote())
-            return ray.get(refs)
+            return ray.get(refs)[0]
```

Comment:
Can we add a comment here and in the docstring as to why we are only returning the 0th worker's state?
We are probably assuming they are always the same anyways due to cross-GPU synching prior to gradient computations ??

Reply:
updated the code. Hope it's more clear now.